lib/segment/src/types.rs

```diff
@@ -197,6 +198,13 @@ impl PartialEq for ScoredPoint {
     }
 }
 
+impl Hash for ScoredPoint {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.id.hash(state);
+        self.version.hash(state);
+    }
+}
```

Comment:
I would not recommend doing that, it might have unpredictable behavior if somebody tries to use it some other context.
For example we would want to implement read consistency using this hash - what is going to happen if id & version are same, but payload is different? The answer is - nobody knows.

Reply:
That's true, I guess a new type or wrapper would be better suited here